docs/adrs/0003-authorization-bypass-for-ui-autotests.md

```diff
@@ -0,0 +1,29 @@
+# 3. Authorization bypass for UI auto tests
+
+Date: 2023-10-26
+
+## Status
+
+Accepted
+
+## Context
+
+Up until now, we have been unable to run ui automation tests on dev or test enviroments.
+
+Due to DfE's policy of only allowing certain IP ranges access to authenticate via microsoft, test users will not work in the test pipelines in the same way as a normal user so we needed to implement a special route to enable the test user to bypass authentication in non-prod environments.
+
+## Decision
+
+We will implement the dotnet class DenyAnonymousAuthorizationRequirement which allow bypassing of authorization functionality.
+
+It requires a secret guid to be stored in azure which is passed in an authorization header with every test automation request.
+
+The bypass is only allowed on non production environments.
+
+This method as been used in other Dfe projects within the program.
+
+## Consequences
```

Comment:
More consequences:

- Developers will no longer be able to access a locally running CI docker instance through their browser because the header will be missing from their browser requests
- Integration and Deployment tests can be reenabled in the pipeline. The security test pipeline will also now function on demand